### DISCUSSION TITLE
Libre 2 production is in danger in Germany, we need to have Libre 3 support!

### DISCUSSION THREAD

**Discussion by pooyaww** [NightscoutFoundation/xDrip#2051] [General]
According to a letter I received this morning from Abbott, it seems a competitor company in Germany has started a patent war and it is possible that Abbott stops providing Libre 2 forever! They are already sending sensor packages earlier to people having insurance coverage before the court forbid them to sell Libre 2. It seems they can only continue selling Libre 3 in near future and asking people to switch to Libre 3 as soon as possible.
We need Xdrip+ for Libre 3.
![abott](https://user-images.githubusercontent.com/9465942/158623241-f3c8925e-1bb9-4bdd-83f1-5cbaf123d016.jpg)

**Comment by yoshimo** [comment]
For rooted devices there might be a way soon to read the data from the official app.

**Comment by Navid200** [comment]
@tzachi-dar Any ideas or suggestions?

**Comment by tzachi-dar** [comment]
A few people are working on this.
Right now a solution will probably require a rooted phone.

**Reply by ConsFBL** [reply]
Latest information for the German market: the libre 3 reader will be launched in 3rd quarter 2022, so could be as early as 1.7.22. Right after, all (German) users will be force-transferred to libre 3, and libre 2 will be discontinued. So I am sorry to say this is really getting urgent.

**Comment by Teute0815** [comment]
Wouldn't it be possible to add
LibreView and/or
LibreLinkUp
accounts as possible Data sources in xDrip in order to receive the values?

That was not feasible with L2, as that didn't read/send the values itself without scanning, but the L3 does that automatically every minute.

Even though it requires internet connection it would be a first/quick and easy to use "workaround" until there's a proper/easy (offline) solution

Atm there are a lot of users who can't/don't want to use the other available solutions - of which I found three so far:
*rooted phone - "XPosed/LSPosed Modul für Libre 3"
*LibreLinkUp "collector" apk
*Nightscout script/plugin - Follower

**Reply by ANDREAPIPPI** [reply]
Hi.
Libre 3 will be available in Italy in the third quarter of this year, too.
May be the legal problems involved with Libre 2 are more serious than supposed, because Libre 3  for Italy was scheduled in 2023, at best.
So I support the request by Teute0815, this is becoming a real need for a lot of people that use or will soon use Libre 3.
Thank you guys for all what you have done, are doing and will do ....   
Andrea  :-)

**Reply by pooyaww** [reply]
> Wouldn't it be possible to add LibreView and/or LibreLinkUp accounts as possible Data sources in xDrip in order to receive the values?
> 
> That was not feasible with L2, as that didn't read/send the values itself without scanning, but the L3 does that automatically every minute.
> 
> Even though it requires internet connection it would be a first/quick and easy to use "workaround" until there's a proper/easy (offline) solution

I think it is possible but there might be a problem. In this case we should relay on the values that are already modified and calibrated via LibreLink app and it makes problem with the xDrip calibration algorithms which I think it modifies the raw data. But if the aim is just having the readings in xDrip and using it as an interface it can be a quick workaround.

**Reply by Teute0815** [reply]
As the Libre App is just really bad and rudimentary any workaround to be able to still use xDrip would be much appreciated.

**Comment by wefink** [comment]
hope u find a solution for libre3 soon....and thanks for your work.
w

**Comment by dassystem** [comment]
It's happening now (in Germany). My subscription from Libre 2 has been changed to Libre 3 and the first batch of 3 sensors is on its way to me. 

Unfortunately there is still no (non-root) offline way to connect Libre3 with xDrip. I found this [repo](https://github.com/maheini/FreeStyle-Libre-3-patch), but at this moment it is more a collection of information than working software.

**Reply by ConsFBL** [reply]
Diabox states they are on their way to offer an offline way to read the libre 3 data for looping (seen in Telegram discussion last week). I am definitely looking forward to this, expected release in 2-3 months. To my understanding this seems to be offline with no rooted phone.

**Comment by twinko** [comment]
Hi everybody, i just wanted to know if anyone is reverse engenineering the libre 3 app? A solution like the patched libre 2 app would be awesome. Is there a community anywhere? I couldnt find anything online. Sadly rooting issnt possible for me due to banking apps.

**Comment by ANDREAPIPPI** [comment]
Hello.
FreeStyke 3 is now distributed in Italy by the National Health Service, and I just received my first supply of sensors, installed the app and... wow!
One of the ugliest apps I've ever seen on my phone!
First of all, I don't understand why Abbott insists on that white background layout, which not only requires more energy to run (the black pixels are simply turned off, the white ones are energized and thus suck battery power), but if it comes on at night it wakes up those who are sleeping with us like a beacon;
then the light grey colour of the characters, which with the white background are almost invisible in broad daylight;
the graphics are very poor, without any care;
almost no setting of the alarms, which are (or should be!) the most important feature of a CGM;
impossibility of setting the colours of the screen, which is white and remains white!
If we compare xDrip+ with this app, the former wins out over the latter 10-0.
So, given that Italy is now among the countries with FS 3 in widespread use, and many other countries are taking the same route, what could be done to connect xDrip+ to FS 3?
Root the phone?
Fork xDrip+?
Anything else?
I think that in the near future all FS users will be forced to switch to FS 3, and this is almost impossible with the original application, so much that I would definitely prefer to keep the old FS 1 with Miaomiao and xDrip+!
So, please, help us!
Andrea

**Reply by JulGer** [reply]
At least dark mode is working on Android 12.

**Reply by ANDREAPIPPI** [reply]
> At least dark mode is working on Android 12.

-JulGer, can you please tell me how?
I have a Xiaomi with MIUI Global 13.0.7, based on Android 12.
Thank you.
Andrea

**Comment by Navid200** [comment]
One option is to use xDrip as a follower using the "Web Follower" hardware data source.  This will require internet access.

Has anyone tried "Companion" option under Hardware Data Source for this?  I am asking because I am curious if it works or not.

**Reply by steve8x8** [reply]
Neither Libre 2 nor Libre 3 official (or patched, since the ID wouldn't be changed) apps are yet (as of 2022-12-05) supported by Companion mode, AFAICT.
This would require additions to coOptedPackages (in UiBasedCollector.java): "com.freestylelibre.app.${countrycode}" for 2, "com.freestylelibre3.app.${countrycode}" for 3 (countrycode="de" and likely more, e.g. "en", "es", "nl" etc).

**Comment by Teute0815** [comment]
The easiest/best option atm to link L3 and X-Drip+ without root is the
**LLU client**
from nechoj in my opinion.

The connection works just like L2 and Libre patched app in x-drip+. Only letdown: internet required.
[LLU Client](https://insulinclub.de/index.php?thread/33987-llu-client/&pageNo=1)

I'm still running my last L2s, but I tried the client with two test sensors and it worked nicely.
Still hoping for an unrooted, offline option in the future.

@ANDREAPIPPI 
@Navid200

**Reply by twinko** [reply]
Yeah doesnt sound like a perfect solution and im kind of interested on how heavily this will effect the battery, but it has to do for now i think.

Moreover you have to calculate with about 1,35 gb per month only for the llu client. (good luck if you travel  :/ )
getting the data from the cloud every minute: 45MB per day x 30 days= 1,35 GB

Regards

**Reply by Teute0815** [reply]
you can adjust the timeframe. I set it to 5 minutes, as that's what I'm used to from x-drip anyways.

**Reply by twinko** [reply]
Activate the raw data in xdrip settings, believe me, it changes how you read the graph. Once done you dont want to go back :)
And these come every minute

**Comment by chriske64** [comment]
There seems to be some progress: https://github.com/maheini/FreeStyle-Libre-3-patch/releases/tag/1.1.0

**Reply by maheini** [reply]
New version is released with multilingual support: https://github.com/maheini/FreeStyle-Libre-3-patch

**Comment by twinko** [comment]
@jamorham there is a patched Libre 3_v3.3.0 apk:
https://github.com/maheini/FreeStyle-Libre-3-patch/tree/main/Patched%20Apk

Would be awesome if you could implement it as data source!

**Reply by paulyglocke** [reply]
@jamorham It would be incredible if some update to Xdrip could recognise the patched 3 app, the more stable readings granted to Libre 3 could allow SMB support in AAPS for those who don't have Dexcom, I am also being forced onto Libre 3 now in Germany, if I had a solution as elegant as the Patched L2 App to Xdrip, it would make things much easier for me and many others in Germany looping.  Thank so much, and thanks for all your hard work, dont know what I would have done without your app these last 2 years of looping.

**Reply by steve8x8** [reply]
The app identifies the same as the original, as `com.freestylelibre3.app.de`, fwiw...

**Comment by nigelkeirross** [comment]
I believe that juggluco can read Libre3 sensors. I have not tried this, but juggluco can send results to xDrip+.

**Reply by steve8x8** [reply]
App id: `tk.glucodata` - since I'm not using it, does it have a notification that could serve the Companion mode?

**Comment by ANDREAPIPPI** [comment]
Hello
It's a long time without new infos, nothing happened meantime?
I tried to use FSL 3 App, but it is really ugly and doesn'rt give all the information, data, setting possibilities and graphics that xDrip+ gives.
Any possibility to have an xDrip+ patch for FSL 3?
Please!

**Reply by maheini** [reply]
Have you tried: https://github.com/maheini/FreeStyle-Libre-3-patch/blob/main/Juggluco-solution/instructions/en/instructions.md

Or the new solution: https://github.com/maheini/FreeStyle-Libre-3-patch/discussions/27

(not working for US version. Please consider Diabox for the US version)

**Reply by ANDREAPIPPI** [reply]
Thank you, I had not seen them. 
I'll try it. 
Andrea   :-)

**Reply by maheini** [reply]
Glad to hear you don't already knew them. I think it's exactly what you're searching for (at least for the EU libre version).
